packages/test-utils/src/contract.ts

```diff
@@ -0,0 +1,111 @@
+import * as ethers from "ethers";
+import { HIGH_GAS_LIMIT } from "./utils";
+
+const { solidityKeccak256 } = ethers.utils;
+
+export interface TruffleContract {
+  readonly contractName: string;
+  readonly abi: any[];
+  readonly bytecode: string;
+  readonly networks: { [networkName: string]: { address: string } };
+}
+
+export class Contract extends ethers.Contract {
+  public salt?: string;
+  public cfAddress?: string;
+  public registry?: Contract;
+}
+
+export class AbstractContract {
+  public static loadTruffleArtifact(
+    artifactName: string,
+    links?: { [name: string]: AbstractContract }
+  ): AbstractContract {
+    const truffleContract = artifacts.require(artifactName);
+    return AbstractContract.fromTruffleContract(truffleContract, links);
+  }
+
+  public static fromTruffleContract(
+    truffleContract: TruffleContract,
+    links?: { [name: string]: AbstractContract }
+  ): AbstractContract {
+    let { bytecode } = truffleContract;
+    if (links) {
+      for (const name of Object.keys(links)) {
+        const library = links[name];
+        if (!library.deployedAddress) {
+          throw new Error("Library must have a deployed address");
+        }
+        const regex = new RegExp(`__${name}_+`, "g");
+        const addressHex = library.deployedAddress.replace("0x", "");
+        bytecode = bytecode.replace(regex, addressHex);
```

Comment:
Truffle-contract doesn't expose the linkBytecode util function, so I had to copy the source code. 

I can use the `link` method on the Truffle artifact, but that would couple AbstractContract tightly with TruffleContract, which we don't want.